### PR TITLE
Fix for AWSCLI-1745 for IOT Certificates permissions

### PR DESCRIPTION
**Pull request by rekhajoshm**
Fix for AWSCLI-1745 for IOT Certificates permissions

### CODE REVIEW

**Comment by rekhajoshm** [comment]
working locally, tested.
```
$ aws iot create-keys-and-certificate   --set-as-active   --certificate-pem-outfile cert.pem   --public-key-outfile publicKey.pem   --private-key-outfile privateKey.pem

$ ls -l privateKey.pem publicKey.pem cert.pem
-rw-------  1  ..  1220 Jan 23 12:57 cert.pem
-rw-------  1  ..  1679 Jan 23 12:57 privateKey.pem
-rw-------  1  ..  451 Jan 23 12:57 publicKey.pem
```

**Comment by mtdowling** [comment]
Thanks for the PR. I think we might need to make this a little more flexible. I think a file permission setting could be passed into the `QueryOutFileArgument` instead of setting every file to 600. This allows `QueryOutFileArgument` to remain generic.

File: awscli/customizations/arguments.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 from awscli.arguments import CustomArgument
 import jmespath
-
+import os
```
Comment (mtdowling) [inline awscli/customizations/arguments.py]: Note that `os` is already imported.

**Comment by rekhajoshm** [comment]
Thanks @mtdowling updated for review comments.please review.thanks!

File: awscli/customizations/iot.py [hunk 2]
```diff
@@ -49,4 +49,4 @@ def register_create_keys_from_csr_arguments(session, argument_table, **kwargs):
     argument_table['certificate-pem-outfile'] = QueryOutFileArgument(
         session=session, name='certificate-pem-outfile',
         query='certificatePem',
-        after_call_event='after-call.iot.CreateCertificateFromCsr')
+        after_call_event='after-call.iot.CreateCertificateFromCsr',perm=0o600)
```
Comment (mtdowling) [inline awscli/customizations/iot.py]: Minor style issue: missing a space here after the comma

**Comment by mtdowling** [comment]
Thanks for updating the PR! It looks great. Could you just add a test to `QueryOutFileArgument` that ensures that the file it creates has the permissions that were requested? I think we might only be able to conditionally perform this assertion depending on if we're running on a *nix environment. Also a rebase + squash into a single commit would be awesome.

**Comment by rekhajoshm** [comment]
Thanks @mtdowling squashed for review comments, test.please check.thanks!

**Comment by mtdowling** [comment]
:ship: Thanks for your contribution!

**Pull request merged**